packages/@lwc/engine-dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -49,16 +50,26 @@ export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLE
     const HtmlPrototype = getComponentHtmlPrototype(Ctor);
 
     return class extends HtmlPrototype {
+        isHydrated = false;
         constructor() {
             super();
-            createVM(this, Ctor, {
-                mode: 'open',
-                owner: null,
-                tagName: this.tagName,
-            });
+            if (this.isConnected) {
```

Comment:
I would add a note here to clarify that this `if` block is hit if there's already an un-upgraded element in the DOM with the same tag name. (I know this is just the way `customElements.define` works, but I always forget it and find it surprising. 😄 )

It's also worth noting that none of the element properties are upgraded. [Some frameworks actually do this](https://nolanlawson.com/2021/08/03/handling-properties-in-custom-element-upgrades/) (e.g. Lit and Stencil). Maybe it's something we'll want to do in the future?